spec/cancan/controller_resource_spec.rb

```diff
@@ -185,6 +185,34 @@ class HiddenModel < ::Model; end
         expect(resource.send('resource_params')).to eq(resource: 'params')
       end
     end
+
+    context 'with a build parameters method' do
+      it 'prefers to use the new_params method for initializing instances' do
+        params[:action] = 'new'
+        allow(controller).to receive(:resource_params).and_return(resource: 'params')
+        allow(controller).to receive(:model_params).and_return(model: 'params')
+        allow(controller).to receive(:custom_params).and_return(custom: 'params')
+        allow(controller).to receive(:new_params).and_return(name: 'new')
+        allow(controller).to receive(:create_params).and_return(name: 'create')
+        resource = CanCan::ControllerResource.new(controller)
+        resource.load_resource
+        expect(resource.send(:build_params_method)).to eq :new_params
+        expect(resource.send(:resource_instance).name).to eq 'new'
+      end
+
+      it 'prefers to use the new_params method for initializing instances' do
+        params[:action] = 'create'
+        allow(controller).to receive(:resource_params).and_return(resource: 'params')
+        allow(controller).to receive(:model_params).and_return(model: 'params')
+        allow(controller).to receive(:custom_params).and_return(custom: 'params')
+        allow(controller).to receive(:new_params).and_return(name: 'new')
+        allow(controller).to receive(:create_params).and_return(name: 'create')
+        resource = CanCan::ControllerResource.new(controller)
+        resource.load_resource
+        expect(resource.send(:build_params_method)).to eq :create_params
+        expect(resource.send(:resource_instance).name).to eq 'create'
```

Comment:
a bit too much duplicated code in this test. could you please refactor it? also the description of second `it` block is wrong.